src/common/util_pmem.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * util_pmem.h -- internal definitions for pmem utils
+ */
+
+#ifndef NVML_UTIL_PMEM_H
+#define NVML_UTIL_PMEM_H 1
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "libpmem.h"
+
+/*
+ * util_persist -- flush to persistence
+ */
+static inline void
+util_persist(int is_pmem, const void *addr, size_t len)
+{
+	if (is_pmem)
+		pmem_persist(addr, len);
+	else
+		pmem_msync(addr, len);
+}
+
+/*
+ * util_persist_auto -- flush to persistence
+ */
+static inline void
+util_persist_auto(int is_dev_dax, const void *addr, size_t len)
```

Comment:
Hmm... If you know that "is_dev_dax" is true, you might as well use the util_persist, but if you don't know it, pmem_is_pmem will return true for dev dax anyway, so why bother? It just makes the API difficult to use.